packages/runtime/client/rest/src/contract/endpoints/collection.ts

```diff
@@ -192,12 +199,30 @@ export const GetContractsResponseGuard = assertGuardEqual(
 );
 
 /**
- * Request options for the {@link index.RestClient#createContract | Create contract } endpoint
- * @category Endpoints
+ * Either a Non Merkleized Marlowe Contract or a Merkleized One
+ * @category Endpoint : Build Create Contract Tx
+ */
+export type ContractOrSourceId = Contract | SourceId;
+
+/**
+ * Guard for ContractOrSourceId type
+ * @category Endpoint : Build Create Contract Tx
+ */
+export const ContractOrSourceIdGuard: t.Type<ContractOrSourceId> = t.union([
+  G.Contract,
+  SourceIdGuard,
+]);
+
+/**
+ * Request options for the {@link index.RestClient#buildCreateContractTx | Build Create Contract Tx } endpoint
+ * @category Endpoint : Build Create Contract Tx
  */
-export interface CreateContractRequest {
-  // FIXME: create ticket to add stake address
-  // stakeAddress: void;
+export interface BuildCreateContractTxRequest {
+  /**
+   * The Marlowe Runtime utilizes this Optional field to set a stake address
+   * where to send staking rewards for the Marlowe script outputs of this contract.
```

Comment:
I don't think this sentence is grammatically correct. For PRs that rely heavily on documentation, we can ask @sfoo-iohk to pitch in.

My 2 cents for this is (if I understand the option correctly):

`Marlowe contracts can have staking rewards for the ADA locked in the contract. Use this field to set the recipient address of those rewards`